mediawiki.go

```diff
@@ -122,6 +122,27 @@ type uploadResponse struct {
 	}
 }
 
+type UploadDupError struct {
```

Comment:
Are duplicates/existing files the only warnings returned? If not, this should probably be a generic "Warnings" struct.